manual/manual/tutorials/coreexamples.etex

```diff
@@ -677,6 +677,36 @@ print_expr e; print_newline ();;
 print_expr (deriv e "x"); print_newline ();;
 \end{caml_example}
 
+There is a "printf" function in "Printf" module that allows you to make
```

Comment:
I would rather mention the more featurefull `Format` module.

More importantly, modules are only introduced in the next chapter, a forward reference would be nice. Something like
```latex
(see chapter~\ref{c:moduleexamples})
```

Reply:
Wouldn't it make sense to have a version with the simpler Printf first, and maybe (but not in the present PR) a mention of Format later?

Reply:
This is certainly an option; but since Format features are a superset of Printf, skipping `Printf` works too .

Reply:
Here is why I'm not convinced by this idea:

- `Format` is more complex than `Printf`, because of formatting boxes and the @!%weird syntax they use in format strings. If we don't introduce them, Format adds no value, but if we do introduce them, they need quite a bit more introduction on their own, so much that showing Printf first does not make the whole thing much more cumbersome to read (it's going in increment of complexity).

- Printf is arguably the best choice for format-string printing in the OCaml ecosystem (Jane Street used to have its own printf-style extension, I'm not sure if it's actively used anymore?); it's a part of the language tutorial that users will certain find used in real code. On the other hand, there are quite a few more pretty-printing libraries than Format, some built on top of it, some on entirely different formatting engines, so this part of the tutorial (is nice to have but) would feel less canonical.